dbms/src/Interpreters/executeQuery.cpp

```diff
@@ -200,6 +200,14 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
 
         auto interpreter = InterpreterFactory::get(ast, context, stage);
         res = interpreter->execute();
+        if (InterpreterInsertQuery * insertInterpreter = typeid_cast<InterpreterInsertQuery *>(&*interpreter))
+        {
+            String database;
+            String table_name;
+            insertInterpreter->getDatabaseTable(database, table_name);
+            if (!database.empty())
+                context.setCurrentTable(database, table_name);
```

Comment:
I think that it's not a good idea to put current table into context. maybe, it's better to put it into BlockIO or change return type of executeQuery function (or add another output argument).

Reply:
It's a query context that should be destroyed with query itself.